ethcore/src/state/account.rs

```diff
@@ -206,6 +214,16 @@ impl Account {
 		self.storage_changes = storage;
 	}
 
+	/// Swtich account to the patched mode
+	pub fn switch_to_patched(&mut self) {
+		self.patched_account = true;
```

Comment:
Reseting some account values here could make sense. Maybe this function could be combine with 'patch_account' somehow to avoid strange account state if future code forgot to call 'patch_account'.

Reply:
I was thinking heavily on how to achieve this, because I also don't like the this kind of dangling state for account. Didn't find a good solution. This information (if account is patched) is required both in state and account, but the external code works with the state only. So in such constraints this was the only suitable solution (IMO). 
Meanwhile 'forget' to patch account is pretty difficult, the more dangerous situation may appear, if someone starts to use patched accounts as regular ones. Here I tried to prevent this with asserts in commit methods.

Reply:
I think I am missing some code/context, I only find 'switch_to_patched' function call in 'patch_account' account function, so I was simply thinking of removing this fn and make the bool switch in account fn 'reset_code_and_storage' (which seems to be also called at a single place just after switch_to_patched).